examples/View.elm

```diff
@@ -1,79 +1,165 @@
+import Utils
+import Model exposing (Model, Record, Msg(..), Sort(..))
 
 
 formatLastModified : Int -> Time -> String
 formatLastModified timestamp currentTime =
-    timeAgo (toFloat timestamp) currentTime
+    Utils.timeAgo (toFloat timestamp) currentTime
 
 
-iconBtn : String -> Msg -> Html Msg
+iconBtn : String -> Msg -> Html.Html Msg
 iconBtn icon action =
-    button [ class "btn btn-xs btn-default", onClick action ]
-        [ i [ class ("glyphicon glyphicon-" ++ icon) ] [] ]
+    Html.button
+        [ Html.Attributes.class "btn btn-xs btn-default"
+        , Html.Events.onClick action
+        ]
+        [ Html.i [ Html.Attributes.class ("glyphicon glyphicon-" ++ icon) ]
+            []
+        ]
 
 
-recordRow : Time -> Record -> Html Msg
+recordRow : Time -> Record -> Html.Html Msg
 recordRow currentTime { id, title, description, last_modified } =
-    tr []
-        [ td [] [ text id ]
-        , td [] [ text (Maybe.withDefault "[empty]" title) ]
-        , td [] [ text (Maybe.withDefault "[empty]" description) ]
-        , td [] [ text (formatLastModified last_modified currentTime) ]
-        , td []
+    Html.tr []
+        [ Html.td [] [ Html.text id ]
+        , Html.td [] [ Html.text (Maybe.withDefault "[empty]" title) ]
+        , Html.td [] [ Html.text (Maybe.withDefault "[empty]" description) ]
+        , Html.td [] [ Html.text (formatLastModified last_modified currentTime) ]
+        , Html.td []
             [ iconBtn "edit" (EditRecord id)
-            , text " "
+            , Html.text " "
             , iconBtn "trash" (DeleteRecord id)
             ]
         ]
 
 
-recordsList : List Record -> Time -> Html Msg
-recordsList records currentTime =
-    table [ class "table" ]
-        [ thead []
-            [ tr []
-                [ th [] [ text "id" ]
-                , th [] [ text "title" ]
-                , th [] [ text "description" ]
-                , th [] [ text "last_modified" ]
-                , th [] []
-                ]
+recordHeaders : Sort -> List (Html.Html Msg)
+recordHeaders sort =
+    let
+        ( icon, sortColumn ) =
+            case sort of
+                Asc column ->
+                    ( "glyphicon glyphicon-sort-by-attributes", column )
+
+                Desc column ->
+                    ( "glyphicon glyphicon-sort-by-attributes-alt", column )
+
+        headings =
+            [ "id", "title", "description", "last_modified" ]
+
+        headingWithSortIcon : String -> String -> String -> Html.Html Msg
+        headingWithSortIcon icon sortColumn heading =
+            let
+                content =
+                    if heading == sortColumn then
+                        [ Html.i [ Html.Attributes.class icon ] []
+                        , Html.text <| " " ++ heading
+                        ]
+                    else
+                        [ Html.text heading ]
+            in
+                Html.th [ Html.Events.onClick (SortByColumn heading) ] content
+    in
```

Comment:
Nested `let` statements are hard to read and follow, would it be possible to generally have a single level only in this file?